server/dbs/mongo.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+##############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+##############################################################################
+
+from pymongo import MongoClient
+
+from . import base
+
+
+class MongoConnector(base.DatabaseConnector):
+    name = 'mongo'
+
+    def __init__(self, *args, **kwargs):
+        super(MongoConnector, self).__init__(*args, **kwargs)
+        if not self.validate(**kwargs):
+            return
+
+        self.database = kwargs.get('database')
+        self.collection = kwargs.get('collection')
+        self.databaseUrl = kwargs.get('url')
+
+        self.initialized = True
+
+    def connect(self):
+        self.conn = MongoClient(self.databaseUrl)
+
+    def disconnect(self):
+        self.conn.close()
+        self.conn = None
+
+    def performSelect(self, fields, queryProps={}, filters=[], client=None):
+        print "hiiii"
+        self.connect()
+
+        result = super(MongoConnector, self).performSelect(fields, queryProps, filters)
+        coll = self.conn[self.database][self.collection]
+
+        opts = {}
+        for k, v in queryProps.iteritems():
+            target = None
+            if k == 'fields':
+                target = 'projection'
+                if v == []:
+                    v = None
+            elif k == 'offset':
+                target = 'skip'
+            elif k in ['filter', 'limit', 'no_cursor_timeout', 'cursor_type', 'sort', 'allow_partial_results', 'oplog_replay', 'modifiers']:
+                target = k
+
+            if target is not None:
+                opts[target] = v
+
+        result['data'] = list(coll.find(**opts))
```

Comment:
When returning data as a list, I was expecting a list-of-lists, with column information specified by the fields.  We should, ideally, either make the base class capable of handling both list-of-lists and list-of-dicts automatically, or we should transform this into the expected form.

The list-of-lists format uses vastly less bandwidth, which matters when querying hundreds of thousands of records.